packages/components/pie-button/src/button.scss

```diff
@@ -1,160 +1,128 @@
-@use '@justeat/pie-design-tokens/dist/jet.scss' as dt;
-
-$button-height-xs: 32px;
-$button-height-s: 40px;
-$button-height-m: 48px;
-$button-height-l: 56px;
-
-$button-border-radius: dt.$radius-rounded-e;
-$button-font: dt.$font-interactive-m-family;
-$button-focus: dt.$color-focus;
-
-$button-bg-color-primary: dt.$color-interactive-brand;
-$button-text-color-primary: dt.$color-content-interactive-primary;
-
-$button-bg-color-secondary: dt.$color-interactive-secondary;
-$button-text-color-secondary: dt.$color-content-interactive-secondary;
-
-$button-bg-color-outline: dt.$color-container-default;
-$button-text-color-outline: dt.$color-content-interactive-tertiary;
-$button-outline-color-outline: dt.$color-border-strong;
-
-$button-bg-color-ghost: dt.$color-container-default;
-$button-text-color-ghost: dt.$color-content-link;
-$button-disabled-bg-color-ghost: dt.$color-container-default;
-
-$button-bg-color-disabled: dt.$color-disabled-01;
-$button-text-color-disabled: dt.$color-content-disabled;
-$button-outline-color-disabled: dt.$color-disabled-01;
-
-// Sets up the size of a button such as xsmall or large
-@mixin button-sizing($min-block-size, $padding-block, $padding-inline, $font-size, $font-weight, $line-height) {
-    min-block-size: $min-block-size;
-    padding: $padding-block $padding-inline;
-    font-size: $font-size * 1px;
-    font-weight: $font-weight;
-    line-height: $line-height * 1px;
+:host {
+    --btn-border-radius: var(--dt-radius-rounded-e);
+    --btn-font-family: var(--dt-font-interactive-m-family);
+    --btn-font-weight: var(--dt-font-interactive-m-weight);
+
+    // The base values are set to the size default, which is for the medium button size
+    --btn-padding: 10px var(--dt-spacing-e);
+    --btn-min-height: $button-height-m;
+    --btn-font-size: calc(var(--dt-font-size-20) * 1px);
```

Comment:
Worth noting that we'll look to tidy up these font-size declarations as we start to create a CSS utility library that is shared across components (which will happen shortly)